portalnet/src/overlay_service.rs

```diff
@@ -1282,6 +1282,22 @@ where
                 )
             })?;
 
+        // Attempt to get semaphore permit if fails we return an empty accept
+        let permit = match self
+            .utp_controller
+            .inbound_utp_transfer_semaphore
+            .clone()
+            .try_acquire_owned()
+        {
+            Ok(permit) => permit,
+            Err(_) => {
+                return Ok(Accept {
+                    connection_id: 0,
```

Comment:
Well I guess my understanding was something along the lines that a connection id of `0` was an "empty accept" (according to the comment above). My understanding was that a peer wouldn't try to connect with a connection id of `0`. But, I guess that was incorrect. 

Good find, I just think it's definitely worth adding some comments here explaining why this should happen before we populate the bitlist & that the semaphore check is effectively free